test/relay_client_test.js

```diff
@@ -450,26 +451,49 @@ contract('RelayClient', function (accounts) {
     })
 
 
-    it("should send relay balance to owner after removed", async function () {
-
-        let response = await request(localhostOne+'/getaddr');
-        let relayServerAddress = JSON.parse(response.body).RelayServerAddress;
-        let beforeOwnerBalance = await web3.eth.getBalance(relayOwner);
-        let res = await rhub.removeRelayByOwner(relayServerAddress, {from:relayOwner});
-        assert.equal("RelayRemoved", res.logs[0].event);
-        assert.equal(relayServerAddress.toLowerCase(), res.logs[0].args.relay.toLowerCase());
-
-        let i = 0;
-        let relayBalance = await web3.eth.getBalance(relayServerAddress);
-        while (relayBalance != 0 && i < 20) {
-            await testutils.sleep(200);
-            relayBalance = await web3.eth.getBalance(relayServerAddress);
-            i++
-        }
-        assert.equal(0,relayBalance)
-        let afterOwnerBalance = await web3.eth.getBalance(relayOwner);
-        assert.equal(true,parseInt(afterOwnerBalance)  > parseInt(beforeOwnerBalance))
+    describe("relay balance management", async function () {
+        let relayServerAddress;
+        let beforeOwnerBalance;
+        it("should NOT send relay balance to owner after removed", async function () {
```

Comment:
Essentially, you wait here 4 seconds to verify the relay does not send out it's balance...

Can't you test it with go unittest?

Reply:
I can, but this is not really a unit test, so decided to keep it in js.

Reply:
I am not sure js tests should cover server behavior.

Reply:
This was already there. I think that as this one is more like integration tests so it's not the end of the world having these tests in js.